File: erlang/sum-of-multiples/sum_of_multiples.erl
-module(sum_of_multiples).
-export([sumOfMultiplesDefault/1, sumOfMultiples/2]).

sumOfMultiplesDefault(N) -> sumOfMultiples([3,5], N).

sumOfMultiples(_, 0) -> 0;
sumOfMultiples(Multiples, N) ->
  NaturalNumber = N - 1,
  case lists:any(fun(Multiple) -> NaturalNumber rem Multiple == 0 end, Multiples) of
    true  -> NaturalNumber + sumOfMultiples(Multiples, NaturalNumber);
    false -> 0 + sumOfMultiples(Multiples, NaturalNumber)
  end.

File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize(*multiples)
    @multiples = multiples
  end

  def to(limit)
    limit.times.select { |i| is_multiple?(i) }.reduce(:+)
  end

  def self.to(limit)
    new(3,5).to(limit)
  end

  private

  def is_multiple?(i)
    @multiples.find { |multiple| i % multiple == 0 }
  end
end
